pkg/certmanager/certmanager.go

```diff
@@ -48,6 +51,8 @@ type Options struct {
 
 	// IssuerRef is used as the issuerRef on created CertificateRequests.
 	IssuerRef cmmeta.ObjectReference
+
+	AdditionalAnnotations map[string]string
```

Comment:
Add comment. Also add validation to ensure that all key values acceptable as annotations (https://github.com/kubernetes/apimachinery/blob/6b1428efc73348cc1c33935f3a39ab0f2f01d23d/pkg/api/validation/objectmeta.go#L44)

Reply:
We just need to use that function linked, and return error and exit error the program if the given annotations aren't permissible.

Reply:
Yes, I wasn't sure where to perform that exit, but I think I've found an appropriate place now.